runtime/jcl/common/shared.c

```diff
@@ -1261,7 +1286,9 @@ Java_com_ibm_oti_shared_SharedClassURLHelperImpl_findSharedClassImpl3(JNIEnv* en
 	if (!getStringPair(env, &nameChars, &nameLen, &partitionChars, &partitionLen, classNameObj, partitionObj)) {
 		goto _errorPostClassNamePartition;
 	}
-	if (!(cpeType = getCpeTypeForProtocol((char*)urlElements.protocolChars, urlElements.protocolLen, urlElements.pathChars, urlElements.pathLen))) {
+	char* correctPath = NULL;
+	cpeType = getCpeTypeForProtocol(env, (char*)urlElements.protocolChars, urlElements.protocolLen, urlElements.pathChars, urlElements.pathLen, &correctPath);
```

Comment:
As we already have the corrected path here, we can pass it to `getCachedURL()` below with an additional parameter. `getCachedURL()` does not need to call `correctURLPath()` to calculate the corrected path again.